src/modules/checkIsValidUserInput.js

```diff
@@ -9,7 +9,23 @@
  * @return {boolean} - True if the user input is valid, false otherwise
  */
 function checkIsValidUserInput(userInput) {
-  /* Write your code here */
+  if (userInput.length !== 4) {
+    return false;
+  }
+
+  if (userInput[0] === '0') {
+    return false;
+  }
+
+  const arr = [...userInput];
+
+  for (const num of arr) {
+    if (arr.filter((number) => number === num).length > 1 || isNaN(+num)) {
```

Comment:
The current logic checks for duplicate digits by filtering the array for each digit and checking the length. This approach is inefficient. Consider using a Set to track seen digits, which would simplify the logic and improve performance.